NHapi20/NHapi.Base/Model/GenericMessage.cs

```diff
@@ -66,6 +66,30 @@ public static Type getGenericMessageClass(String version)
 			{
 				c = typeof (V25);
 			}
+			else if (version.Equals("2.5.1"))

```

Comment:
Would have probably been easier to use a dictionary with the string version as the key and the type as the value.